### DISCUSSION TITLE
I have a question: will support for Intel VT-X/AMD-V hardware virtualization and NVIDIA CUDA/AMD GPUFORT acceleration be added?

### DISCUSSION THREAD

**Discussion by Burinis** [86Box/86Box#3671] [Ideas]
As I understand it, these technologies improve the performance of programs. But will they be supported?

**Comment by DDXofficial** [comment]
No.

86Box is an **emulator**, not a hypervisor/virtualization software like VirtualBox and VMware. Internally, it does not perform the same tasks as a hypervisor (i.e. allocating your computer's cores, hardware virtualization, etc.) and instead emulates an ENTIRE PC (i.e. CPU, memory, graphics card, sound card, etc.) from the 80s-90s in software. There is no amount of hardware virtualization/acceleration that can improve emulator performance here - your only bet would be to look into processors with high single-threaded performance.

No such "modern" features such as hardware virtualization may be added to the project in the foreseeable future.

**Comment by cold-brewed** [comment]
Some further information is available in [the FAQ](https://86box.readthedocs.io/en/latest/usage/faq.html#what-is-the-difference-between-86box-and-applications-like-virtualbox-or-virtual-pc).

**Comment by dhrdlicka** [comment]
There are no plans currently to support either. 

It is also worth noting that neither of these technologies are magic boxes to "improve the performance of programs". They _can_ do that, but only if they are used appropriately.

Intel VT-X and AMD-V would greatly impact the nature of the emulator and is unlikely to be done. It is not _impossible_ to use it as the CPU core and then emulate the other parts of the system, such as the chipset, expansion cards and such in software, in fact that's what other virtualizers already do, albeit usually with custom hardware that is not constrained by compatibility with existing drivers. Virtual PC or QEMU emulates some real hardware, but the range is not as large as 86Box.

The problem is that it is optimized for modern software and comes with a great performance penalty when generally just doing stuff in a way that modern operating systems don't. For example, the developer of NTVDM-x64 tried doing a HAXM driver and found out that the VGA performance is [terrible](https://github.com/leecher1337/ntvdmx64/issues/150). Guest software might also break due to the VM running faster than it was designed for.

CUDA in specific might be interesting to look into for GPU emulation, as currently the host CPU does all the heavy lifting. Using CUDA might help move some of that workload onto the GPU on compatible hosts.
